certbot-nginx/certbot_nginx/configurator.py

```diff
@@ -565,6 +566,45 @@ def _has_certbot_redirect(self, vhost, domain):
         test_redirect_block = _test_block_from_block(_redirect_block_for_domain(domain))
         return vhost.contains_list(test_redirect_block)
 
+    def _should_break_block(self, vhost):
```

Comment:
This particular set of conditions seems to be particularly for hsts, let's either make that more clear so it's not accidentally used elsewhere, or just put the `if` inline.